external_search.py

```diff
@@ -1103,85 +1159,6 @@ def _match_range(cls, field, operation, value):
 class Query(SearchBase):
     """An attempt to find something in the search index."""
 
-    # When we run a simple query string search, we are matching the
-    # query string against these fields.
-    SIMPLE_QUERY_STRING_FIELDS = [
-        # These fields have been stemmed.
-        'title^4',
-        "series^4",
-        'subtitle^3',
-        'summary^2',
-        "classifications.term^2",
-
-        # These fields only use the standard analyzer and are closer to the
-        # original text.
-        'author^6',
-        'publisher',
-        'imprint'
-    ]
-
-    # When we look for a close match against title, author, or series,
-    # we apply minimal stemming (or no stemming, in the case of the
-    # author), because we're handling the case where the user typed
-    # something in exactly as is.
-    #
-    # TODO: If we're really serious about 'minimal stemming', we
-    # should use a version that doesn't stem plurals or remove stop
-    # words.
-    MINIMAL_STEMMING_QUERY_FIELDS = [
-        'title.minimal', 'author', 'series.minimal'
-    ]
-
-    # When we run a fuzzy query string search, we are matching the
-    # query string against these fields. It's more important that we
-    # use fields that have undergone minimal stemming because the part
-    # of the word that was stemmed may be the part that is misspelled
-    FUZZY_QUERY_STRING_FIELDS = [
-        'title.minimal^4',
-        'series.minimal^4',
-        "subtitle.minimal^3",
-        "summary.minimal^2",
-        'author^4',
-        'publisher',
-        'imprint'
-    ]
-
-    # These words will fuzzy-match other common words that aren't relevant,
-    # so if they're present and correctly spelled we shouldn't use a
-    # fuzzy query.
-    FUZZY_CONFOUNDERS = [
-        "baseball", "basketball", # These fuzzy match each other
-
-        "soccer", # Fuzzy matches "saucer", "docker", "sorcery"
-
-        "football", "softball", "software", "postwar",
-
-        "tennis",
-
-        "hamlet", "harlem", "amulet", "tablet",
-
-        "biology", "ecology", "zoology", "geology",
-
-        "joke", "jokes" # "jake"
-
-        "cat", "cats",
-        "car", "cars",
-        "war", "wars",
-
-        "away", "stay",
-    ]
```

Comment:
We now "spell-check" the search result, and if it looks like everything is spelled correctly the fuzzy hypotheses are given a much lower weight. So there's no need to check for specific words.

I added integration tests of some of these fuzzy confounders to the circulation branch, to verify that any future solutions don't reintroduce this problem.